prog.go

```diff
@@ -42,34 +47,55 @@ type ProgramSpec struct {
 // Program represents a Program file descriptor
 type Program struct {
 	fd       uint32
+	name     string
 	progType ProgType
 }
 
+var (
+	detectHaveProgName sync.Once
+	haveProgName       bool
+)
+
 // NewProgram creates a new Program.
 func NewProgram(spec *ProgramSpec) (*Program, error) {
 	if len(spec.Instructions) == 0 {
-		return nil, fmt.Errorf("instructions cannot be empty")
+		return nil, errors.Errorf("instructions cannot be empty")
 	}
-	buf := bytes.NewBuffer(make([]byte, 0, len(spec.Instructions)*asm.InstructionSize))
-	err := spec.Instructions.Marshal(buf, nativeEndian)
+
+	detectHaveProgName.Do(func() {
```

Comment:
Black magic! We should document that creating a program can cause more than one to be loaded :P

Reply:
Will do.